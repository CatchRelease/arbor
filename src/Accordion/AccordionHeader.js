import React from 'react';
import PropTypes from 'prop-types';

import StyledAccordionHeader from './StyledAccordionHeader';
import StyledAccordionHeading from './StyledAccordionHeading';
import StyledAccordionHeadingButton from './StyledAccordionHeadingButton';
import Text from '../Text';

const AccordionHeaderNote = ({ note }) =>
  note && (
    <Text fontSize="size4" textAlign="right" color="text.muted">
      {note}
    </Text>
  );

const AccordionHeader = ({ isOpen, note, onClick, panelId, text }) => (
  <StyledAccordionHeader
    alignItems="center"
    justifyContent="space-between"
    p="regular"
  >
    <StyledAccordionHeading fontWeight="medium" mr="regular">
      <StyledAccordionHeadingButton
        type="button"
        aria-controls={panelId}
        aria-expanded={isOpen}
        onClick={onClick}
      >
        {text}
      </StyledAccordionHeadingButton>
    </StyledAccordionHeading>
    <AccordionHeaderNote note={note} />
  </StyledAccordionHeader>
);

AccordionHeader.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  note: PropTypes.node,
  onClick: PropTypes.func,
  panelId: PropTypes.string.isRequired,
  text: PropTypes.node.isRequired
};

AccordionHeader.defaultProps = {
  note: null,
  onClick: () => {}
};

export default AccordionHeader;
