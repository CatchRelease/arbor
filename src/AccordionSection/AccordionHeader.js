import PropTypes from 'prop-types';

import StyledAccordionHeader from './StyledAccordionHeader';
import StyledAccordionHeading from './StyledAccordionHeading';
import StyledAccordionHeadingText from './StyledAccordionHeadingText';
import Icon from '../Icon';
import VARIANTS from './variants';

const AccordionHeaderIcon = ({ isOpen, ...props }) => (
  <Icon mr="small" name="chevron" rotation={isOpen ? null : '270'} {...props} />
);

AccordionHeaderIcon.propTypes = {
  isOpen: PropTypes.bool.isRequired
};

const AccordionHeader = ({
  isOpen,
  note,
  onClick,
  panelId,
  text,
  iconRight,
  ...props
}) => (
  <StyledAccordionHeader
    alignItems="center"
    justifyContent="space-between"
    px="regular"
    onClick={onClick}
    aria-controls={panelId}
    aria-expanded={isOpen}
    {...props}
  >
    <StyledAccordionHeading fontWeight="medium" mr="regular">
      <StyledAccordionHeadingText
        alignItems="center"
        aria-controls={panelId}
        aria-expanded={isOpen}
      >
        {iconRight ? (
          <>
            {text}
            <AccordionHeaderIcon isOpen={isOpen} ml="smallest" />
          </>
        ) : (
          <>
            <AccordionHeaderIcon isOpen={isOpen} />
            {text}
          </>
        )}
      </StyledAccordionHeadingText>
    </StyledAccordionHeading>
    {note}
  </StyledAccordionHeader>
);

AccordionHeader.propTypes = {
  /**
   * Whether or not the AccordionSection this header resides in is open or not.
   */
  isOpen: PropTypes.bool.isRequired,

  /**
   * Optional note to render within the AccordionHeader
   */
  note: PropTypes.node,

  /**
   * Callback to call whenever the StyledAccordionHeadingButton is clicked
   */
  onClick: PropTypes.func,

  /**
   * ID of the AccordionPanel that holds the content of the AccordionSection.
   * Used to specify aria-controls for accessbility.
   */
  panelId: PropTypes.string.isRequired,

  /**
   * Header text for the section
   */
  text: PropTypes.node.isRequired,

  /**
   * Link variant for styling. See storybook for examples.
   */
  variant: PropTypes.oneOf(VARIANTS).isRequired,

  /**
   * Optional flag to position icon to right of header text
   */
  iconRight: PropTypes.bool
};

AccordionHeader.defaultProps = {
  note: null,
  onClick: () => {},
  iconRight: false
};

export default AccordionHeader;
