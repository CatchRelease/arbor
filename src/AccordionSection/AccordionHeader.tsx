import { FC, MouseEventHandler, ReactNode } from 'react';

import StyledAccordionHeader from './StyledAccordionHeader';
import StyledAccordionHeading from './StyledAccordionHeading';
import StyledAccordionHeadingText from './StyledAccordionHeadingText';

import AccordionHeaderIcon from './AccordionHeaderIcon';
import { Variant } from './variants';

type Props = {
  iconRight?: boolean;
  isOpen: boolean | undefined;
  note?: ReactNode;
  onClick?: MouseEventHandler<HTMLElement>;
  panelId: string;
  text: ReactNode;
  variant?: Variant;
};

const AccordionHeader: FC<Props> = ({
  isOpen,
  note,
  onClick = () => {},
  panelId,
  text,
  iconRight = false,
  ...props
}) => (
  <StyledAccordionHeader
    alignItems="center"
    justifyContent="space-between"
    px="regular"
    onClick={onClick}
    aria-controls={panelId}
    aria-expanded={isOpen}
    {...props}
  >
    <StyledAccordionHeading fontSize="size4" fontWeight="medium" mr="regular">
      <StyledAccordionHeadingText
        alignItems="center"
        aria-controls={panelId}
        aria-expanded={isOpen}
      >
        {iconRight ? (
          <>
            {text}
            <AccordionHeaderIcon isOpen={isOpen} ml="smallest" />
          </>
        ) : (
          <>
            <AccordionHeaderIcon isOpen={isOpen} />
            {text}
          </>
        )}
      </StyledAccordionHeadingText>
    </StyledAccordionHeading>
    {note}
  </StyledAccordionHeader>
);

export default AccordionHeader;
