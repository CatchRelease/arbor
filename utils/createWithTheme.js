import React from 'react';
import { CacheProvider } from '@emotion/core';
import createCache from '@emotion/cache';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';
import theme from '../src/theme';

const myCache = createCache();
myCache.compat = true;

export default (Component) =>
  create(
    <CacheProvider value={myCache}>
      <ThemeProvider theme={theme}>{Component}</ThemeProvider>
    </CacheProvider>
  );
