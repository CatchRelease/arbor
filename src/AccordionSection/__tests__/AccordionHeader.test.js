import createWithTheme from '../../../utils/createWithTheme';
import AccordionHeader from '../AccordionHeader';

describe('<AccordionHeader />', () => {
  const baseProps = {
    variant: 'default',
    isOpen: false,
    panelId: 'my-accordion-1',
    text: 'My Heading'
  };

  const createTree = (additionalProps = {}) =>
    createWithTheme(
      <AccordionHeader {...{ ...baseProps, ...additionalProps }} />
    );

  it('properly renders an accordion header', () => {
    const tree = createTree();

    expect(tree).toMatchSnapshot();
  });

  it('renders with a note', () => {
    const tree = createTree({
      note: 'My Accordion Section Note'
    });

    expect(tree).toMatchSnapshot();
  });

  it('renders with the panel visible', () => {
    const tree = createTree({ isOpen: true });

    expect(tree).toMatchSnapshot();
  });

  it('renders with an on click', () => {
    const tree = createTree({ onClick: jest.fn() });

    expect(tree).toMatchSnapshot();
  });

  it('renders with chevron postion', () => {
    const tree = createTree({ chevronRight: true });

    expect(tree).toMatchSnapshot();
  });
});
