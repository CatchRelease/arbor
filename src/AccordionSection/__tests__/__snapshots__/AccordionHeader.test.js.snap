// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionHeader /> properly renders an accordion header 1`] = `
.emotion-6 {
  box-sizing: border-box;
  border-bottom: 1px solid #E0E1E2;
  padding-left: 1rem;
  padding-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #F2F2F2;
}

section:last-child .emotion-6 {
  border-bottom: none;
}

.emotion-6 .button--minimal:last-of-type {
  margin-right: -0.25rem;
}

.emotion-6 .button--minimal:last-of-type +:not(object) {
  margin-left: 0.25rem;
}

.emotion-6 button + .button--secondary,
.emotion-6 button + .button--primary {
  margin-left: 0.5rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
  margin-right: 1rem;
  line-height: 1.25;
  margin-bottom: calc(0.875rem * 0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  cursor: pointer;
  font-size: inherit;
  padding: 0;
}

.emotion-0 {
  margin-right: 0.75rem;
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

<header
  className="emotion-6 emotion-7"
>
  <h1
    className="emotion-4 emotion-5"
    color="text.dark"
    fontSize="size4"
    fontWeight="medium"
  >
    <button
      aria-controls="my-accordion-1"
      aria-expanded={false}
      className="emotion-2 emotion-3"
      onClick={[Function]}
      type="button"
    >
      <i
        className="ar ar-caret emotion-0 emotion-1"
      />
      My Heading
    </button>
  </h1>
</header>
`;

exports[`<AccordionHeader /> renders with a note 1`] = `
.emotion-6 {
  box-sizing: border-box;
  border-bottom: 1px solid #E0E1E2;
  padding-left: 1rem;
  padding-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #F2F2F2;
}

section:last-child .emotion-6 {
  border-bottom: none;
}

.emotion-6 .button--minimal:last-of-type {
  margin-right: -0.25rem;
}

.emotion-6 .button--minimal:last-of-type +:not(object) {
  margin-left: 0.25rem;
}

.emotion-6 button + .button--secondary,
.emotion-6 button + .button--primary {
  margin-left: 0.5rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
  margin-right: 1rem;
  line-height: 1.25;
  margin-bottom: calc(0.875rem * 0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  cursor: pointer;
  font-size: inherit;
  padding: 0;
}

.emotion-0 {
  margin-right: 0.75rem;
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

<header
  className="emotion-6 emotion-7"
>
  <h1
    className="emotion-4 emotion-5"
    color="text.dark"
    fontSize="size4"
    fontWeight="medium"
  >
    <button
      aria-controls="my-accordion-1"
      aria-expanded={false}
      className="emotion-2 emotion-3"
      onClick={[Function]}
      type="button"
    >
      <i
        className="ar ar-caret emotion-0 emotion-1"
      />
      My Heading
    </button>
  </h1>
  My Accordion Section Note
</header>
`;

exports[`<AccordionHeader /> renders with an on click 1`] = `
.emotion-6 {
  box-sizing: border-box;
  border-bottom: 1px solid #E0E1E2;
  padding-left: 1rem;
  padding-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #F2F2F2;
}

section:last-child .emotion-6 {
  border-bottom: none;
}

.emotion-6 .button--minimal:last-of-type {
  margin-right: -0.25rem;
}

.emotion-6 .button--minimal:last-of-type +:not(object) {
  margin-left: 0.25rem;
}

.emotion-6 button + .button--secondary,
.emotion-6 button + .button--primary {
  margin-left: 0.5rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
  margin-right: 1rem;
  line-height: 1.25;
  margin-bottom: calc(0.875rem * 0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  cursor: pointer;
  font-size: inherit;
  padding: 0;
}

.emotion-0 {
  margin-right: 0.75rem;
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

<header
  className="emotion-6 emotion-7"
>
  <h1
    className="emotion-4 emotion-5"
    color="text.dark"
    fontSize="size4"
    fontWeight="medium"
  >
    <button
      aria-controls="my-accordion-1"
      aria-expanded={false}
      className="emotion-2 emotion-3"
      onClick={[MockFunction]}
      type="button"
    >
      <i
        className="ar ar-caret emotion-0 emotion-1"
      />
      My Heading
    </button>
  </h1>
</header>
`;

exports[`<AccordionHeader /> renders with the panel visible 1`] = `
.emotion-6 {
  box-sizing: border-box;
  border-bottom: 1px solid #E0E1E2;
  padding-left: 1rem;
  padding-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #F2F2F2;
}

section:last-child .emotion-6 {
  border-bottom: none;
}

.emotion-6 .button--minimal:last-of-type {
  margin-right: -0.25rem;
}

.emotion-6 .button--minimal:last-of-type +:not(object) {
  margin-left: 0.25rem;
}

.emotion-6 button + .button--secondary,
.emotion-6 button + .button--primary {
  margin-left: 0.5rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
  margin-right: 1rem;
  line-height: 1.25;
  margin-bottom: calc(0.875rem * 0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  cursor: pointer;
  font-size: inherit;
  padding: 0;
}

.emotion-0 {
  margin-right: 0.75rem;
}

<header
  className="emotion-6 emotion-7"
>
  <h1
    className="emotion-4 emotion-5"
    color="text.dark"
    fontSize="size4"
    fontWeight="medium"
  >
    <button
      aria-controls="my-accordion-1"
      aria-expanded={true}
      className="emotion-2 emotion-3"
      onClick={[Function]}
      type="button"
    >
      <i
        className="ar ar-caret emotion-0 emotion-1"
      />
      My Heading
    </button>
  </h1>
</header>
`;
